src/internal/ppsutil/util.go

```
@@ -433,7 +433,7 @@ func GetStatsCommit(commitInfo *pfs.CommitInfo) *pfs.Commit {
 			return commitRange.Lower
 		}
 	}
-	// TODO: Getting here would be a bug in 2.0, log?
+	// TODO: Getting here would be a bug in 2.0, log? -- worth discussion?
```

Ah, this is just false because of spouts, right?

---

it seems like we do sometimes hit this. I'm trying to understand when that would happen

---

I think we can change this comment to better reflect reality (that this might sometimes be called for spouts or services)

---

We shouldn't be calling this function for services / spouts. I think we should either return an error or panic when it isn't found so we don't end up using a nil commit when we shouldn't.